File: Queue/Queue-Java/src/Queue/Queue.java
package Queue;

public class Queue
{
  public Queue()
  {
    this.rear=null;
    this.front=null;
    this.size=0;
  }

  public void enqueue(String word)
  {
    Node newNode=new Node(new Data(word));
    if(isEmpty())
    {
      this.rear=newNode;
      this.front=newNode;
    }
    else 
    {
      this.rear.setNext(newNode);
      this.rear=newNode;
    }
    this.size++;
  }

  public void dequeue()
  {
    if(isEmpty())
    {
      return;
    }

    this.front=this.front.getNext();
    if(this.front==null)
    {
      this.rear=null;
    }
    this.size--;
  }

  public void peek()
  {
    if(isEmpty())
    {
      return;
    }

    System.out.println(this.front.getData().getData());
  }

  public void view()
  {
    if(isEmpty())
    {
      return;
    }

    Node current=this.front;
    while(current!=null)
    {
      System.out.println(current.getData().getData());
      current=current.getNext();
    }
  }

  public int getSize()
  {
    return this.size;
  }

  private boolean isEmpty()
  {
    return this.front==null;
  }

  private Node rear;
  private Node front;
  private int size;
}

File: Queue/Queue-Python/src/Queue.py
from Node import Node
from Data import Data

class Queue:
    def __init__(self):
        self.rear=None
        self.front=None
        self.size=0

    def enqueue(self,word):
        newNode=Node(Data(word))
        if self.__isEmpty():
            self.rear=newNode
            self.front=newNode
        else:
            self.rear.next=newNode
            self.rear=newNode
        self.size+=1

    def dequeue(self):
        if self.__isEmpty():
            return
        self.front=self.front.next
        if self.front is None:
            self.rear=None
        self.size-=1

    def size(self):
        return self.size

    def view(self):
        if self.__isEmpty():
            return

        current=self.front
        while current is not None:
            print(current.data.word)
            current=current.next

    def __isEmpty(self):
        return self.front is None
